packages/components/src/unit-control/index.tsx

```diff
@@ -71,11 +76,26 @@ function UnforwardedUnitControl(
 	// still passes `null` as a `value`.
 	const nonNullValueProp = valueProp ?? undefined;
 	const [ units, reFirstCharacterOfUnits ] = useMemo( () => {
-		const list = getUnitsWithCurrentUnit(
+		let list = getUnitsWithCurrentUnit(
 			nonNullValueProp,
 			unitProp,
 			unitsProp
 		);
+		// If allowedUnitValues is provided, we need to filter the list of units.
+		if ( allowedUnitValues?.length ) {
```

Comment:
So this is a huge hack and just here to illustrate the proposal.

The goal is for this component to honor the order of units. For example, if I wanted to have `%` appear at the top of the unit list.

I'm just not sure whether to do it in the component or as part of `useCustomUnits`.

https://github.com/WordPress/gutenberg/pull/31822#discussion_r633280823 suggests that `useCustomUnits` might be the wrong place to do it.

Reply:
cc @ciampo for advice - non-urgent, when you get time 🙇🏻